lib/server/services/qualityControlFlag/QcFlagService.js

```diff
@@ -118,16 +115,19 @@ class QcFlagService {
             });
             if (!targetRun) {
                 // eslint-disable-next-line max-len
-                throw new BadParameterError(`You cannot insert flag for data pass (id:${dataPassId}), run (runNumber:${runNumber}), detector (name:${dplDetector.name}) as there is no association between them`);
+                throw new BadParameterError(`There is not association between data pass with this id (${dataPassId}), run with this number (${runNumber}) and detector with this name (${dplDetector.name})`);
+            }
+
+            const fromTime = from ?? targetRun.startTime;
+            const toTime = to ?? targetRun.endTime;
+
+            if (fromTime >= toTime) {
+                throw new BadParameterError('Parameter "to" timestamp must be greater than "from" timestamp');
             }
 
-            const isFromTimeBeyondReange = fromTime
-                && (fromTime < targetRun.timeTrgStart.getTime() || targetRun.timeTrgEnd.getTime() < fromTime);
-            const isToTimeBeyondReange = toTime
-                && (toTime < targetRun.timeTrgStart.getTime() || targetRun.timeTrgEnd.getTime() < toTime);
-            if (isFromTimeBeyondReange || isToTimeBeyondReange) {
+            if (fromTime < targetRun.startTime || targetRun.endTime < toTime) {
```

Comment:
I think here what you want is:
```suggestion
            if (targetRun.endTime < fromTime || toTime < targetRun.startTime) {
```
Wrong if run ended before flag start, or if flag ended before run start